lib/awful/tag.lua

```diff
@@ -465,6 +465,7 @@ function tag.object.set_screen(t, s)
 
     -- Change the screen
     tag.setproperty(t, "screen", s)
+    tag.setproperty(t, "index", #s:get_tags(true))
```

Comment:
I'm not sure if this matters much, but perhaps the index should be updated before the screen is changed so that the index is already correct when `property::screen` occurs?